sleekxmpp/plugins/xep_0313/mam.py

```diff
@@ -21,6 +22,130 @@
 log = logging.getLogger(__name__)
 
 
+
+
+class Archive_Query:
+    def __init__(self, xmpp, start=None, end=None, with_jid=None, continue_after=None,
+                 timeout=None, callback=None, number_of_queried_elements=None, collect_all=True):
+        self._xmpp = xmpp
+        self.timeout = timeout
+        self.collect_all = collect_all
+        self._callback_ptr = callback
+        
+        self._collector = None
+        self._cleanup_callback = None
+        
+        self._query_id = None
+        self._iq = None
+        self._last_message = None
+        
+    
+        ## prepare query
+        iq = self._xmpp.Iq()
+        self._query_id = iq['id']
+
+        iq['type'] = 'set'
+        iq['mam']['queryid'] = self._query_id
+        iq['mam']['start'] = start
+        iq['mam']['end'] = end
+        iq['mam']['with'] = with_jid
+        if ( number_of_queried_elements ):
+            iq['mam']['rsm']['max'] = str(number_of_queried_elements)
```

Comment:
It would make sense to modify the rsm plugin to automatically convert to and from ints, see sleekxmpp/plugins/xep_0047/stanza.py for an example.